Client/core/agent.py

```
@@ -0,0 +1,108 @@
+class Agent:
+    def __init__(self, config):
+        self.config = config
+        self.model_loader = ModelLoader(config)
+        self.memory_manager = MemoryManager()
+        self.action_executor = ActionExecutor()
+        self.info_text = load_info()
+        self.context_builder = ContextBuilder(self.info_text, self.memory_manager)
+        self.running = False
+
+    def start(self):
+        logging.info("Starting Agent...")
```

Ruff (0.15.0)

[warning] `info()` call on root logger

(LOG015)